src/JSObjectValuesProxy.cc

```diff
@@ -0,0 +1,151 @@
+  PyObject_GC_Del(self);
+}
+
+Py_ssize_t JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_length(JSObjectValuesProxy *self)
+{
+  if (self->dv.dv_dict == NULL) {
+    return 0;
+  }
+  return JSObjectProxyMethodDefinitions::JSObjectProxy_length((JSObjectProxy *)self->dv.dv_dict);
+}
+
+int JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_contains(JSObjectValuesProxy *self, PyObject *key)
+{
+  if (self->dv.dv_dict == NULL) {
+    return 0;
+  }
+  return JSObjectProxyMethodDefinitions::JSObjectProxy_contains((JSObjectProxy *)self->dv.dv_dict, key);
+}
+
+int JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_traverse(JSObjectValuesProxy *self, visitproc visit, void *arg) {
+  Py_VISIT(self->dv.dv_dict);
+  return 0;
+}
+
+// private
+static int all_contained_in(PyObject *self, PyObject *other) {
+  PyObject *iter = PyObject_GetIter(self);
+  int ok = 1;
+
+  if (iter == NULL) {
+    return -1;
+  }
+
+  for (;; ) {
+    PyObject *next = PyIter_Next(iter);
+    if (next == NULL) {
+      if (PyErr_Occurred())
+        ok = -1;
+      break;
+    }
+    if (PyObject_TypeCheck(other, &JSObjectValuesProxyType)) {
+      JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_contains((JSObjectValuesProxy *)other, next);
+    }
+    else {
+      ok = PySequence_Contains(other, next);
+    }
+    Py_DECREF(next);
+    if (ok <= 0)
+      break;
+  }
+
+  Py_DECREF(iter);
+  return ok;
+}
+
+PyObject *JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_iter(JSObjectValuesProxy *self) {
+  JSObjectIterProxy *iterator = PyObject_GC_New(JSObjectIterProxy, &JSObjectIterProxyType);
+  if (iterator == NULL) {
+    return NULL;
+  }
+  iterator->it.reversed = false;
+  iterator->it.it_index = 0;
+  iterator->it.kind = KIND_VALUES;
+  Py_INCREF(self->dv.dv_dict);
+  iterator->it.di_dict = self->dv.dv_dict;
+  iterator->it.props = new JS::RootedIdVector(GLOBAL_CX);
+  // Get **enumerable** own properties
+  if (!js::GetPropertyKeys(GLOBAL_CX, ((JSObjectProxy *)(self->dv.dv_dict))->jsObject, JSITER_OWNONLY, iterator->it.props)) {
+    return NULL;
+  }
+  PyObject_GC_Track(iterator);
+  return (PyObject *)iterator;
+}
+
+PyObject *JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_iter_reverse(JSObjectValuesProxy *self) {
+  JSObjectIterProxy *iterator = PyObject_GC_New(JSObjectIterProxy, &JSObjectIterProxyType);
+  if (iterator == NULL) {
+    return NULL;
+  }
+  iterator->it.reversed = true;
+  iterator->it.it_index = JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_length(self) - 1;
+  iterator->it.kind = KIND_VALUES;
+  Py_INCREF(self->dv.dv_dict);
+  iterator->it.di_dict = self->dv.dv_dict;
+  iterator->it.props = new JS::RootedIdVector(GLOBAL_CX);
+  // Get **enumerable** own properties
+  if (!js::GetPropertyKeys(GLOBAL_CX, ((JSObjectProxy *)(self->dv.dv_dict))->jsObject, JSITER_OWNONLY, iterator->it.props)) {
+    return NULL;
+  }
+  PyObject_GC_Track(iterator);
+  return (PyObject *)iterator;
```

Comment:
Similar here.
```suggestion
  JSObjectIterProxy *iterator = JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_iter(self);
  iterator->it.reversed = true;
  iterator->it.it_index = JSObjectValuesProxyMethodDefinitions::JSObjectValuesProxy_length(self) - 1;
  return (PyObject *)iterator;
```

Reply:
this suggestion does not seem to do it all